lib/dockerregistry/inventory.go

```diff
@@ -1952,6 +1952,20 @@ func MKPopulateRequestsForPromotionEdges(
 	}
 }
 
+// RunChecks runs input PullChecks in order to check the promotion.
+func (sc *SyncContext) RunChecks(
+	edges map[PromotionEdge]interface{},
+	pullChecks []PullCheck,
+) error {
+	for _, pullCheck := range pullChecks {
+		err := pullCheck.Run(edges)
+		if err != nil {
+			return err
```

Comment:
You should collect all errors and bubble up all failures, not just the first one to fail.